addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6365,6 +6376,43 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
         const optQuality = uiFragment.querySelector('we-range[data-set-quality]');
         optQuality.setAttribute('data-name', 'quality_range_opt');
     },
+    /**
+     * @private
+     */
+    _updateFilterAvailability() {
+        if (!this.el) {
+            return;
+        }
+        let filterWidget = this.el.querySelector('we-select[data-name="glfilter_select_opt"]');
+        if (!filterWidget) {
+            filterWidget = [...this.el.querySelectorAll("we-select")].find(select =>
+                select.querySelector("we-button[data-gl-filter]")
+            );
+            if (filterWidget) {
+                filterWidget.dataset.name = "glfilter_select_opt";
+            }
+        }
+        if (!filterWidget) {
+            return;
+        }
+        const tooltipTarget = filterWidget.querySelector("we-toggler") || filterWidget;
+        const tooltip = _t("WebGL is not enabled on your browser.");
+        if (!isWebGLEnabled()) {
+            filterWidget.dataset.disabled = "true";
+            filterWidget.classList.add("o_we_disabled_widget");
+            filterWidget.setAttribute("aria-disabled", "true");
+            tooltipTarget.setAttribute("title", tooltip);
+            tooltipTarget.dataset.webglDisabledTooltip = tooltip;
+        } else {
+            delete filterWidget.dataset.disabled;
+            filterWidget.classList.remove("o_we_disabled_widget");
+            filterWidget.removeAttribute("aria-disabled");
+            if (tooltipTarget.dataset.webglDisabledTooltip === tooltip) {
+                tooltipTarget.removeAttribute("title");
+                delete tooltipTarget.dataset.webglDisabledTooltip;
+            }
```

Comment:
Maybe you could just check if `dataset.webglDisabledTooltip` before removing everything:

```suggestion
            if (tooltipTarget.dataset.webglDisabledTooltip) {
                delete tooltipTarget.dataset.webglDisabledTooltip;
                filterWidget.removeAttribute("aria-disabled");
                tooltipTarget.removeAttribute("title");
            }
```
That way, you no longer need to store `tooltip` in a variable outside the first `if`.